HadoopPositionOutputStream.close() is not idempotent (i.e. calling it more than once may fail)

### Describe the bug, including details regarding any error messages, version, and platform.

### Describe the bug, including details regarding any error messages, version, and platform.
If HadoopPositionOutputStream.close() is called more than once, an exception may be thrown. Since HadoopPositionOutputStream implements Closable, its close method should have no effect if already closed.

Example exception:

```
Caused by: org.apache.parquet.util.AutoCloseables$ParquetCloseResourceException: Unable to close resource
	at org.apache.parquet.util.AutoCloseables.uncheckedClose(AutoCloseables.java:85) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.util.AutoCloseables.uncheckedClose(AutoCloseables.java:94) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.hadoop.InternalParquetRecordWriter.close(InternalParquetRecordWriter.java:144) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.hadoop.ParquetRecordWriter.close(ParquetRecordWriter.java:223) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.hadoop.hive.ql.io.parquet.write.ParquetRecordWriterWrapper.close(ParquetRecordWriterWrapper.java:137) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
...
Caused by: java.nio.channels.ClosedChannelException
	at org.apache.hadoop.hdfs.ExceptionLastSeen.throwException4Close(ExceptionLastSeen.java:73) ~[hadoop-hdfs-client-3.4.1.jar:?]
	at org.apache.hadoop.hdfs.DFSOutputStream.checkClosed(DFSOutputStream.java:158) ~[hadoop-hdfs-client-3.4.1.jar:?]
	at org.apache.hadoop.hdfs.DFSOutputStream.flushOrSync(DFSOutputStream.java:639) ~[hadoop-hdfs-client-3.4.1.jar:?]
	at org.apache.hadoop.hdfs.DFSOutputStream.hflush(DFSOutputStream.java:594) ~[hadoop-hdfs-client-3.4.1.jar:?]
	at org.apache.hadoop.fs.FSDataOutputStream.hflush(FSDataOutputStream.java:136) ~[hadoop-common-3.4.1.jar:?]
	at org.apache.parquet.hadoop.util.HadoopPositionOutputStream.close(HadoopPositionOutputStream.java:65) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.hadoop.ParquetFileWriter.close(ParquetFileWriter.java:1663) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.util.AutoCloseables.close(AutoCloseables.java:49) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.util.AutoCloseables.uncheckedClose(AutoCloseables.java:83) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.util.AutoCloseables.uncheckedClose(AutoCloseables.java:94) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.hadoop.InternalParquetRecordWriter.close(InternalParquetRecordWriter.java:144) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.parquet.hadoop.ParquetRecordWriter.close(ParquetRecordWriter.java:223) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
	at org.apache.hadoop.hive.ql.io.parquet.write.ParquetRecordWriterWrapper.close(ParquetRecordWriterWrapper.java:137) ~[hive-exec-4.0.0-20-IBM.jar:4.0.0]
```

### Component(s)
parquet-hadoop

### Component(s)

_No response_

--- a/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopPositionOutputStream.java
+++ b/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopPositionOutputStream.java
@@ -25,6 +25,7 @@ import org.apache.parquet.io.PositionOutputStream;
 
 public class HadoopPositionOutputStream extends PositionOutputStream {
   private final FSDataOutputStream wrapped;
+  private boolean closed;
 
   HadoopPositionOutputStream(FSDataOutputStream wrapped) {
     this.wrapped = wrapped;
@@ -61,8 +62,13 @@ public class HadoopPositionOutputStream extends PositionOutputStream {
 
   @Override
   public void close() throws IOException {
+    if (closed) {
+      return;
+    }
     try (FSDataOutputStream fdos = wrapped) {
       fdos.flush();
+    } finally {
+      closed = true;
     }
   }
 }

--- a/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopPositionOutputStream.java
+++ b/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopPositionOutputStream.java
@@ -20,12 +20,13 @@
 package org.apache.parquet.hadoop.util;
 
 import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
 import org.apache.hadoop.fs.FSDataOutputStream;
 import org.apache.parquet.io.PositionOutputStream;
 
 public class HadoopPositionOutputStream extends PositionOutputStream {
   private final FSDataOutputStream wrapped;
-  private boolean closed;
+  private final AtomicBoolean closed = new AtomicBoolean(false);
 
   HadoopPositionOutputStream(FSDataOutputStream wrapped) {
     this.wrapped = wrapped;
@@ -62,13 +63,11 @@ public class HadoopPositionOutputStream extends PositionOutputStream {
 
   @Override
   public void close() throws IOException {
-    if (closed) {
+    if (closed.getAndSet(true)) {
       return;
     }
     try (FSDataOutputStream fdos = wrapped) {
       fdos.flush();
-    } finally {
-      closed = true;
     }
   }
 }

